### ISSUE TITLE
What support should a maintainer/project expect from prometheus-community?

### ISSUE THREAD

**Issue by RichiH** [prometheus-community/prometheus-community#4]
Do we want to offer anything specific?

What immediately comes to mind is access to infrastructure for testing and such.

**Comment by simonpasquier** [comment]
Yes I see that a consistent and shared CI infrastructure makes it easier to keep up with the changes to the CI provider(s) as it has been experienced with GitHub integrations, Circle CI 1.0 → 2.0 and so on.

Also:
* Dedicated website?
* Binaries should be at least available from the GitHub release pages. Do they deserve a dedicated page similar to https://prometheus.io/download/? 
* Container image hosting?
* Common tooling for releasing artifacts (part of CI probably).

**Comment by carlpett** [comment]
It was mentioned either in the mail thread or in the other issue here that major changes to "how things are done" could also be something which is done in a sweep over everything in this org (for example as the switch to Kingpin on the main org was done, and the upcoming suggested TLS implementations)

**Comment by roidelapluie** [comment]
- regular vendor upgrade and releases

**Comment by carlpett** [comment]
@roidelapluie Isn't that going to be the job of the maintainers? Doing vendor updates and releases requires some amount of care and knowledge of the workings of the exporter, in my mind, rather than just bulk updating everything.

**Comment by roidelapluie** [comment]
That is untrue.

Vendor updates takes mainly good unit and acceptance testing.

**Comment by roidelapluie** [comment]
Plus everyone is a maintainer in my vision.

**Comment by carlpett** [comment]
I'd agree that a good acceptance test coverage would reduce risk. However, I think this is lacking in a significant majority of exporters? I wouldn't say creating them is simple either, in most cases it would require setting up a fair chunk of infrastructure and exercising (which may in some cases also have a non-trivial financial impact).

**Comment by brian-brazil** [comment]
> However, I think this is lacking in a significant majority of exporters?

The JMX exporter is the only one I know of with testing to this level. Some of the push-ones like influxdb/statsd/graphite probably also count. (Node exporter has extensive testing, but I'm not counting it here as it's running against fake data rather than a real kernel, and doesnt cover all OSes either). 

Updating something like Kingpin can be done in bulk, updating libraries related to an exporter's key role less so. It's the same issue we have with SD and notifiers.
